sensing/src/main/java/com/google/android/sensing/impl/SensingEngineImpl.kt

```diff
@@ -179,18 +187,21 @@ internal class SensingEngineImpl(
 
     // Step 1: Delete db records
     database.deleteRecordsInCapture(captureId)
-    // Step 2: delete the captureFolder
-    val captureFile = File(context.filesDir, captureInfo.captureFolder)
-    val parentFile = captureFile.parentFile
-    val deleted: Boolean
-    withContext(Dispatchers.IO) {
-      deleted = captureFile.deleteRecursively()
-      // delete Participant's folder if there are no data
-      if (parentFile?.list()?.isEmpty() == true) {
-        parentFile.delete()
+    if (isRecapture) {
```

Comment:
Question: Why is the data deleted under the folders only in case of recapture?

Reply:
Because in case of recapture , captured media is being stored at cacheDir, else at filesDir. so only in case of recapture we have to delete folders at filesDir.
Reference : in CaptureViewModel.kt -> 
private val internalStorageFolder: File
    get() =
      if (captureInfo.recapture == true) getApplication<Application>().cacheDir
      else getApplication<Application>().filesDir

Reply:
As discussed, can you please verify if the `FilesDir` is not containing the older entries as well that are captured with previous try.